java-checks/src/main/java/org/sonar/java/checks/SubClassStaticReferenceCheck.java

```diff
@@ -99,7 +99,7 @@ public void visitClass(ClassTree tree) {
     @Override
     public void visitIdentifier(IdentifierTree tree) {
       Type type = tree.symbolType();
-      if (!classType.equals(type) && type.isSubtypeOf(classType)) {
+      if (!classType.is(type.fullyQualifiedName()) && type.isSubtypeOf(classType)) {
```

Comment:
Given that classType is (surprise) the type of a class ( https://github.com/SonarSource/sonar-java/pull/1495/commits/a93673052cd6a7d49956b269e986800ec78024bb#diff-5a39a775425a584a6a9e7a42296dadb0R50 )  I struggle to see the point of that fix :  Do you have a failing test case ?